Android migration guide 2.x to 3.x (#2321)
getsentry/sentry-docs: src/platforms/android/migration.mdx

```java
Sentry.setExtra("extra", "thing");
```

## Migrating from `sentry-android 2.x` to `sentry-android 3.x`

**Package changes**

> The package `io.sentry.core` has been renamed to `io.sentry`, do a String replace on all the usages from `io.sentry.core.` to `io.sentry.` that it should compile normally.
> 
> Other than that, the APIs didn't change. Release heath is enabled by default though, if you prefer not to track the health of your releases, disable it with:

Author: ```suggestion
The package `io.sentry.core` has been renamed to `io.sentry`. To compile correctly, replace the strings on all the usages from `io.sentry.core.` to `io.sentry.`

Other than that, the APIs didn't change. Release Heath sets by default. If you prefer not to track the health of your releases, disable it with:
```

```java
Sentry.setExtra("extra", "thing");
```

## Migrating from `sentry-android 2.x` to `sentry-android 3.x`

> The APIs are pretty much the same, Release heath is enabled by default though, to disable it:

Reviewer: ```suggestion
The APIs haven't changed. However, Release Health sets by default. 

To disable Release Health default:
```

> ## Migrating from `sentry-android 2.x` to `sentry-android 3.x`
> 
> The APIs are pretty much the same, Release heath is enabled by default though, to disable it:
> 
> ```xml
> <application>
>   <meta-data android:name="io.sentry.session-tracking.enable" android:value="false" />
> </application>
> ```
> 
> **Package changes**
> 
> The package `io.sentry.core` has been renamed to `io.sentry`, do a String replace on all the usages from `io.sentry.core.` to `io.sentry.` that it should compile normally.

Reviewer 2: ```suggestion
## Migrating from `sentry-android 2.x` to `sentry-android 3.x`

**Package changes**

The package `io.sentry.core` has been renamed to `io.sentry`, do a String replace on all the usages from `io.sentry.core.` to `io.sentry.` that it should compile normally.

Other than that, the APIs didn't change. Release heath is enabled by default though, if you prefer not to track the health of your releases, disable it with:

```xml
<application>
  <meta-data android:name="io.sentry.session-tracking.enable" android:value="false" />
</application>
```

```

Reviewer: How about this:

The package `io.sentry.core` has been renamed to `io.sentry`. To compile correctly, replace the strings on all the usages from `io.sentry.core.` to `io.sentry`.

Other than that, the APIs didn't change. Release Heath sets by default. If you prefer not to track the health of your releases, disable it with:
